pkg/reconciler/pipelinerun/affinity_assistant.go

```diff
@@ -68,6 +70,37 @@ func (c *Reconciler) createAffinityAssistants(ctx context.Context, wb []v1beta1.
 				if err == nil {
 					logger.Infof("Created StatefulSet %s in namespace %s", affinityAssistantName, namespace)
 				}
+			// check whether the affinity assistant (StatefulSet) exists and the affinity assistant pod is created
+			// this check requires the StatefulSet to have the readyReplicas set to 1 to allow for any delay between the StatefulSet creation
+			// and the necessary pod creation, the delay can be caused by any dependency on PVCs and PVs creation
+			// this case addresses issues specified in https://github.com/tektoncd/pipeline/issues/6586
+			case err == nil && a != nil && a.Status.ReadyReplicas == 1:
+				if nodes == nil {
+					nodes, err = c.KubeClientSet.CoreV1().Nodes().List(ctx, metav1.ListOptions{
+						FieldSelector: "spec.unschedulable=true",
+					})
+					if err != nil {
+						errs = append(errs, fmt.Errorf("could not get the list of nodes, err: %w", err))
+					}
+				}
+				for _, n := range nodes.Items {
+					// get the pod created for a given StatefulSet
+					// pod is assigned ordinal of 0 with the replicas set to 1
+					p, err := c.KubeClientSet.CoreV1().Pods(pr.Namespace).Get(ctx, a.Name+"-0", metav1.GetOptions{})
```

Comment:
I'm a bit confused how this reduces the number of API calls-- it looks like we're now getting the same pod once per unschedulable node? I don't think the logic to get the pod and delete it needs to be inside the for loop of nodes; it seems like the main goal was just to only issue the pod API call if there are unschedulable nodes.

Maybe you could do something like:
```
if len(unschedulableNodes) > 0 {
   // Get pod
   // See if the pod's node is unschedulable (could check for membership in a set of unschedulable nodes that have already been fetched)
   // delete pod if so
}
```

Reply:
+1, look like the get pod calls can be moved out of the for loop?

Reply:
looks like a result of coding on a Saturday after a bunch of iterations 🤣 

I have updated it to make sure `Get` pod call is only done if `len(unschedulableNodes) > 0`.